js_modules/dagit/packages/core/src/assets/LaunchAssetExecutionButton.tsx

```diff
@@ -274,6 +301,41 @@ async function stateForLaunchingAssets(
   };
 }
 
+function getCommonJob(assets: LaunchAssetExecutionAssetNodeFragment[], preferredJobName?: string) {
+  const everyAssetHasJob = (jobName: string) => assets.every((a) => a.jobNames.includes(jobName));
+  const jobsInCommon = assets[0] ? assets[0].jobNames.filter(everyAssetHasJob) : [];
+  return jobsInCommon.find((name) => name === preferredJobName) || jobsInCommon[0];
```

Comment:
Makes me itchy to return the `[0]` item on an array that may intentionally be empty.

Reply:
Hmm let me make this a bit cleaner before I merge 👍